connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -573,11 +562,13 @@ public Void call() throws Exception {
                     return null;
                 }
 
-                if (worker.ownsConnector(connName)) {
+                if (assignment.connectors().contains(connName)) {
                     try {
                         worker.stopConnector(connName);
-                        startConnector(connName);
-                        callback.onCompletion(null, null);
+                        if (startConnector(connName))
```

Comment:
@ewencp in the REST request handling bits, it's checking the retval and throwing an error so it propagates up in the REST response.

`processConnectorConfigUpdates` is called via the herder thread's `tick()` so we don't want to propagate errors up, and exceptions are being logged by `WorkerConnector.startConnector()` so there isn't any need to check the retval there.